File: sdks/typescript/examples/basic.ts
import { createPaycrestClient } from "../src/index.js";

async function main() {
  const apiKey = process.env.PAYCREST_API_KEY || "YOUR_API_KEY";
  const client = createPaycrestClient({ apiKey });
  const sender = client.sender();

  const stats = await sender.getStats();
  console.log(stats);
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});

File: sdks/rust/examples/basic.rs
use paycrest_sdk::{client::ClientOptions, PaycrestClient};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let client = PaycrestClient::new_with_options(ClientOptions {
        sender_api_key: Some(std::env::var("PAYCREST_SENDER_API_KEY")?),
        ..Default::default()
    });

    let sender = client.sender()?;
    let stats = sender.get_stats().await?;
    println!("stats = {:?}", stats);
    Ok(())
}
